ocs_ci/utility/utils.py

```diff
@@ -5483,24 +5501,34 @@ def _generate_files():
                 f.flush()
                 os.fsync(f.fileno())  # Ensure data is written to disk
 
-            yield filepath
+            file_queue.put(filepath)
+        except Exception as e:
+            log.error(f"Error creating file: {e}")
+
+    def file_generator():
+        """Generator that yields file paths from the queue as they are created"""
+        files_received = 0
 
-    # Create folder
-    folder_name = "".join(random.choices(string.ascii_lowercase + string.digits, k=8))
-    folder_path = os.path.join("/tmp", folder_name)
-    os.makedirs(folder_path, exist_ok=True)
+        # Start the executor in a background thread to avoid blocking
+        executor = ThreadPoolExecutor(max_workers=max_workers)
+        futures = [executor.submit(create_file, i) for i in range(num_files)]
 
-    # Generate files using generator
-    created_files = set(_generate_files())
+        # Yield files as they become available
+        while files_received < num_files and not stop_event.is_set():
+            try:
+                # Wait for a file to be available with a timeout to check the stop event
+                filepath = file_queue.get(timeout=0.1)
+                files_received += 1
+                yield filepath
+            except Exception:
```

Comment:
If I understand correctly, we can use a specific exception here that would clarify this part a bit: 
```suggestion
            except queue.Empty:
```